src/core/qgsqueryresultmodel.cpp

```diff
@@ -0,0 +1,115 @@
+/***************************************************************************
+  qgsqueryresultmodel.cpp - QgsQueryResultModel
+
+ ---------------------
+ begin                : 24.12.2020
+ copyright            : (C) 2020 by Alessandro Pasotti
+ email                : elpaso@itopen.it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgsqueryresultmodel.h"
+
+QgsQueryResultModel::QgsQueryResultModel( const QgsAbstractDatabaseProviderConnection::QueryResult &queryResult, QObject *parent )
+  : QAbstractListModel( parent )
+  , mQueryResult( queryResult )
+  , mColumns( queryResult.columns() )
+  , mRowCount( mQueryResult.fetchedRowCount() )
+{
+  if ( mQueryResult.hasNextRow() )
+  {
+    mWorker = new QgsQueryResultFetcher( &mQueryResult );
+    mWorker->moveToThread( &mWorkerThread );
+    connect( &mWorkerThread, &QThread::finished, mWorker, &QObject::deleteLater );
```

Comment:
Delete later won't null the pointer, so this could cause a crash in the destructor

Reply:
Good catch. Would it be ok to move worker's `deleteLater` at the end of the destructor after waiting for the thread to quit?

Reply:
I'd change the raw mWorker  pointer member to be a QPointer instead -- that should help